lib/segment/src/types.rs

```diff
@@ -2756,6 +2756,17 @@ impl Condition {
             | Condition::HasVector(_) => 0,
         }
     }
+
+    pub fn targeted_key(&self) -> Option<PayloadKeyType> {
```

Comment:
Can't we return a reference here?

Reply:
I see. We _could_ make it return a `Cow<PayloadKeyType>` but this is probably not worth the effort.